tests/src/common/dispatcher.rs

```diff
@@ -1,30 +1,35 @@
 use finance::percent::Percent;
 use rewards_dispatcher::{
-    error::ContractError,
-    msg::{ExecuteMsg, InstantiateMsg, QueryMsg, SudoMsg},
+    msg::InstantiateMsg,
     state::reward_scale::{Bar, RewardScale, TotalValueLocked},
 };
-use sdk::{cosmwasm_std::Addr, cw_multi_test::Executor};
+use sdk::cosmwasm_std::Addr;
 
-use crate::common::{ContractWrapper, MockApp};
+use crate::common::test_case::app::App;
 
-use super::ADMIN;
+use super::{CwContractWrapper, ADMIN};
 
-pub struct DispatcherWrapper {
-    contract_wrapper: Box<DispatcherContractWrapper>,
-}
+pub(crate) struct Instantiator;
 
-impl DispatcherWrapper {
+impl Instantiator {
     #[track_caller]
     pub fn instantiate(
-        self,
-        app: &mut MockApp,
+        app: &mut App,
         lpp: Addr,
         oracle: Addr,
         timealarms: Addr,
         treasury: Addr,
     ) -> Addr {
-        let code_id = app.store_code(self.contract_wrapper);
+        // TODO [Rust 1.70] Convert to static item with OnceCell
+        let endpoints: CwContractWrapper<_, _, _, _, _, _, _, _, _, _, _> = CwContractWrapper::new(
```

Comment:
[anywhere] sometimes providing types to references does not look perfect, think about removing it ... after all the type could be easily inferred by looking at the code ;-)